components/org.wso2.carbon.identity.api.idle.account.identification/org.wso2.carbon.identity.api.idle.account.identification.v1/src/main/java/org/wso2/carbon/identity/api/idle/account/identification/v1/core/InactiveUsersManagementApiService.java

```diff
@@ -263,4 +307,80 @@ private void validateDatesCombination(LocalDateTime inactiveAfterDate, LocalDate
                     String.format(error.getDescription()));
         }
     }
+
+    /**
+     * Get the filter node as a list.
+     *
+     * @param filter value of the filter.
+     * @return node tree.
+     * @throws IdleAccountIdentificationClientException Error when validate filters.
+     */
+    private List<ExpressionNode> getExpressionNodes(String filter) throws IdleAccountIdentificationClientException {
+
+        // Filter example : isDisabled eq true.
+        List<ExpressionNode> expressionNodes = new ArrayList<>();
+        FilterTreeBuilder filterTreeBuilder;
+        try {
+            if (StringUtils.isNotBlank(filter)) {
+                filterTreeBuilder = new FilterTreeBuilder(filter);
+                Node rootNode = filterTreeBuilder.buildTree();
+                setExpressionNodeList(rootNode, expressionNodes);
+            }
+        } catch (IOException | IdentityException e) {
+            ErrorMessage error = ErrorMessage.ERROR_INVALID_FILTER;
+            throw new IdleAccountIdentificationClientException(error.getCode(), error.getMessage(),
+                    String.format(error.getDescription()));
+        }
+        return expressionNodes;
+    }
+
+    /**
+     * Set the node values as list of expression.
+     *
+     * @param node       filter node.
+     * @param expression list of expression.
+     * @throws IdleAccountIdentificationClientException Error when passing invalid filter.
+     */
+    private void setExpressionNodeList(Node node, List<ExpressionNode> expression)
+            throws IdleAccountIdentificationClientException {
+
+        if (node instanceof ExpressionNode) {
```

Comment:
We may not need nested if conditions.  First two conditions can be joined.  Also better to return from the top if the conditions fails. Then we can minimize the code shifting to right.

Reply:
This has been addressed.